File: Coding Ninjas - CodeStudio/Basics Of C++ Guided Path/rotate_array.cpp
#include <iostream>
using namespace std;

int main() {
	int n;
    cin >> n;
    int arr[n];
    for(int i = 0; i < n; i++)
        cin >> arr[i];
    int k;
    cin >> k;
    int rotated[n];
    for(int i = 0; i < n; i++) {
        int j = i - k;
        if(j < 0)
            j = n - k + i;
        rotated[j] = arr[i];
    }
    for(int i = 0; i < n; i++) {
        cout << rotated[i] << " ";
    }
    cout << endl;
    return 0;
}

File: Coding Ninjas - CodeStudio/Basics of Java Guided Path/rotate_array.java
import java.util.Scanner;
class Solution {
	
	public static void main(String args[]) {
		
		Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        int k = sc.nextInt();
        
        int rotated[] = new int[n];
        for(int i = 0; i < n; i++) {
            rotated[i] = arr[(i + k) % n];
        }
        
        for(int i = 0; i < n; i++) {
            System.out.print(rotated[i] + " ");
        }
		
	}
}